### PR TITLE
[PATCH v4] linux-gen: scheduler: prepare dynamic load balance

### PR DESCRIPTION
**Pull request by psavol**
Add number of threads and queues per spread statistics as preparation for dynamic spread load balancing.

### CODE REVIEW

File: test/performance/odp_sched_perf.c [hunk 1]
```diff
@@ -232,21 +232,18 @@ static int parse_options(int argc, char *argv[], test_options_t *test_options)
 	num_group = test_options->num_group;
 	num_join  = test_options->num_join;
 	if (num_group > MAX_GROUPS) {
-		printf("Error: Too many groups. Max supported %i.\n",
-		       MAX_GROUPS);
+		printf("Error: Too many groups. Max supported %i.\n", MAX_GROUPS);
```
Comment (MatiasElo) [inline test/performance/odp_sched_perf.c:235]: Could be modified to use `ODPH_ERR()`.

File: test/performance/odp_sched_perf.c [hunk 2]
```diff
@@ -232,21 +232,18 @@ static int parse_options(int argc, char *argv[], test_options_t *test_options)
...
 		ret = -1;
 	}
 
-	if (num_join > num_group) {
-		printf("Error: num_join (%u) larger than num_group (%u).\n",
-		       num_join, num_group);
+	if (num_group > 0 && num_join > num_group) {
+		printf("Error: num_join (%i) larger than num_group (%i).\n", num_join, num_group);
```
Comment (MatiasElo) [inline test/performance/odp_sched_perf.c:240]: Could be modified to use `ODPH_ERR()`.

File: platform/linux-generic/odp_schedule_basic.c [hunk 3]
```diff
@@ -218,7 +218,8 @@ typedef struct {
 	struct {
 		char           name[ODP_SCHED_GROUP_NAME_LEN];
 		odp_thrmask_t  mask;
-		int	       allocated;
```
Comment (MatiasElo) [inline platform/linux-generic/odp_schedule_basic.c:221]: Verb missing from commit subject line.

**Review by MatiasElo — APPROVED** [review]
Couple minor comments but otherwise looking okay.

File: test/performance/odp_sched_perf.c (hunk 1 above)
Comment (psavol) [reply test/performance/odp_sched_perf.c:235]: fixed in v3

File: test/performance/odp_sched_perf.c (hunk 2 above)
Comment (psavol) [reply test/performance/odp_sched_perf.c:240]: fixed in v3

File: platform/linux-generic/odp_schedule_basic.c (hunk 3 above)
Comment (psavol) [reply platform/linux-generic/odp_schedule_basic.c:221]: reworded log text in v3

**Pull request merged**